### ISSUE TITLE
[consistent-type-imports] Support inline `type` imports

### ISSUE THREAD

**Issue by chbdetta** [typescript-eslint/typescript-eslint#4264]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/main/docs/linting/TROUBLESHOOTING.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/consistent-type-imports": ["2", {"prefer": "type-imports"}]
  }
}
```

```TS
import {type ReactElement, Component} from 'react'; // eslint-error
```

**Expected Result**

```
import {type ReactElement, Component} from 'react'; // this should not be an error
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.5.0` |
| `@typescript-eslint/parser`        | `5.5.0` |
| `TypeScript`                       | `4.5.2` |
| `ESLint`                           | `8.3.0` |
| `node`                             | `16.13.0` |

**Comment by bradzacher** [comment]
#4237
will be released next week.
